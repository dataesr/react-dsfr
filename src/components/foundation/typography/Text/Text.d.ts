import * as React from 'react';

export type TextAs = "p" | "span";

export type TextSize = "xs" | "sm" | "md" | "lg" | "lead";

export type TextChildren = string | React.ReactNode;

export type TextClassName = string | Object | any[];

export interface TextProps {
    /**
     * html tag to render
     */
    as?: TextAs;
    size?: TextSize;
    spacing?: string;
    /**
     * If true Spectral is used instead of Marianne.
     */
    alt?: boolean;
    children: TextChildren;
    className?: TextClassName;
}

declare const Text: React.FC<TextProps>;

export default Text;

