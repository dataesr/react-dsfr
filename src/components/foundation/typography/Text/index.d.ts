import type Text from "./Text"

export type { Text }
