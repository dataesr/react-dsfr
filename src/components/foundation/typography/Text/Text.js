import PropTypes from 'prop-types';
import classnames from 'classnames';
import { CLASS_NAME_TYPE } from '../../../../utils/variables';

/**
 *
 * @visibleName Text
 */
const Text = ({
  as, size, alt, className, children,
}) => {
  const HtmlTag = `${as}`;
  return (
    <HtmlTag className={classnames(className, {
      'rf-text--alt': size !== 'lead' && alt,
      [`rf-text--${size}`]: size !== 'md',
      'rf-text': size === 'md',
    })}
    >
      {children}
    </HtmlTag>
  );
};

Text.propTypes = {
  /**
  * html tag to render
  */
  as: PropTypes.oneOf(['p', 'span']),
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'lead']),
  /**
  * If true Spectral is used instead of Marianne.
  */
  alt: PropTypes.bool,
  children: PropTypes.string.isRequired,
  /**
   * One of: string, object
   */
  className: CLASS_NAME_TYPE,
};
Text.defaultProps = {
  as: 'p',
  size: 'md',
  alt: false,
  className: '',
};

export default Text;
