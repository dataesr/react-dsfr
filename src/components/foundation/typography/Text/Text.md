### Texte

```jsx
<Text size="lead">Je suis un texte lead</Text>
```

```jsx
<Text size="lg">Je suis un texte lg</Text>
```

```jsx
<Text size="md">Je suis un texte md</Text>
```

```jsx
<Text size="sm">Je suis un texte sm</Text>
```

```jsx
<Text size="xs">Je suis un texte xs</Text>
```
