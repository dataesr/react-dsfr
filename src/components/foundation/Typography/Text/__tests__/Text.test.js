import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Text from '../Text';

const initialProps = {
  as: 'span',
  size: 'sm',
  alt: true,
  children: 'Text',
};

describe('<Text />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<Text {...initialProps}>{initialProps.children}</Text>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
