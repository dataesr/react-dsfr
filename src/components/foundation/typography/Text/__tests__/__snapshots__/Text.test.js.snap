// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<span
  className="rf-text--alt rf-text--sm"
>
  Text
</span>
`;
